pkg/compose/secrets.go

```diff
@@ -31,7 +32,23 @@ import (
 func (s *composeService) injectSecrets(ctx context.Context, project *types.Project, service types.ServiceConfig, id string) error {
 	for _, config := range service.Secrets {
 		file := project.Secrets[config.Source]
-		if file.Environment == "" {
+		content := file.Content
+
+		if file.Environment != "" {
+			env, ok := project.Environment[file.Environment]
+			if !ok {
+				return fmt.Errorf("environment variable %q required by file %q is not set", file.Environment, file.Name)
+			}
+			content = env
+		} else if file.File != "" {
+			data, err := os.ReadFile(file.File)
```

Comment:
`file` can actually be a folder. Need to change logic here to produce tar, not just extract content